frontend/device/remarkable/device.lua

```diff
@@ -125,5 +126,161 @@ function Remarkable:reboot()
     os.execute("systemctl reboot")
 end
 
+local function osCapture(cmd, raw)
+    local f = io.popen(cmd, 'r')
+    if not f then
+        return nil
+    end
+    local s = f:read('*a')
+    f:close()
+    if not s then
+        return nil
+    end
+    if raw then
+        return s
+    end
+    s = string.gsub(s, '^%s+', '')
+    s = string.gsub(s, '%s+$', '')
+    s = string.gsub(s, '[\n\r]+', ' ')
+    return s
+end
+
+local function getNetworkProperty(path, name)
+    path = string.sub(path, string.len("/codes/eeems/oxide1/") + 1)
+    local json, err = rapidjson.decode(osCapture("rot --object Network:" .. path .. " wifi get " .. name))
+    return json
+end
+local function getBSSProperty(path, name)
+    path = string.sub(path, string.len("/codes/eeems/oxide1/") + 1)
+    local json, err = rapidjson.decode(osCapture("rot --object BSS:" .. path .. " wifi get " .. name))
+    return json
+end
+local function getWifiProperty(name)
+    local json, err = rapidjson.decode(osCapture("rot wifi get " .. name))
+    return json
+end
+local function isempty(s)
+  return s == nil or s == ''
+end
+
+-- wireless
+function Remarkable:initNetworkManager(NetworkMgr)
+    if isempty(osCapture("rot")) then
```

Comment:
if `rot` is an executable and its path is always the same I would like to check if exists using `util.fileExists("/path/to/rot")` and perform the check once at device init. This makes possible to to decide which wireless backend is going to be used at runtime.

Reply:
Or just os.execute("which rot")

1 = doesn't exists, 0 = exists